artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java

```diff
@@ -211,12 +211,13 @@ public void init(AMQPSessionContext protonSession, SASLResult saslResult) throws
                                                            true, //boolean xa,
                                                            (String) null, this, true, operationContext, manager.getPrefixes(), manager.getSecurityDomain());
       } else {
+         String validatedUser = manager.getServer().validateUser(user, passcode, protonSPI.getProtonConnectionDelegate(), manager.getSecurityDomain());
```

Comment:
Why did this become necessary? Auth should already have happened by now, seems messy doing it again.

EDIT: so essentially the same comment as Domenicos in other areas.

Reply:
it should not be duplicated, just earlier. I need to investigate the duplication

Reply:
The same call to validateUser has been remove from manager.getServer().createSession..